crates/core/src/surfnet/locker.rs

```diff
@@ -1605,6 +1620,42 @@ impl SurfnetSvmLocker {
             }
         });
     }
+
+    /// Resets an account in the SVM state
+    ///
+    /// This function coordinates the reset of accounts by calling the SVM's reset_account method.
+    /// It handles program accounts (including their program data accounts) and can optionally
+    /// cascade the reset to all accounts owned by a program.
+    pub fn reset_account(&self, pubkey: Pubkey, config: ResetAccountConfig) -> SurfpoolResult<()> {
+        let cascade_to_owned = config.cascade_to_owned.unwrap_or_else(|| false);
+
+        self.with_svm_writer(move |svm_writer| {
+            if let Some(account) = svm_writer.get_account(&pubkey) {
+                // Check if this is an executable account (program)
+                if account.executable {
+                    // Handle upgradeable program - also reset the program data account
+                    if account.owner == solana_sdk_ids::bpf_loader_upgradeable::id() {
+                        let program_data_address =
+                            solana_sdk::bpf_loader_upgradeable::get_program_data_address(&pubkey);
+
+                        // Reset the program data account first
+                        svm_writer.reset_account(&program_data_address)?;
+                    }
+
+                    if cascade_to_owned {
+                        let owned_accounts = svm_writer.get_account_owned_by(pubkey);
+                        for (owned_pubkey, _) in owned_accounts {
+                            // Avoid infinite recursion by not cascading further
+                            svm_writer.reset_account(&owned_pubkey)?;
+                        }
+                    }
+                }
+                // Reset the account itself (this handles both regular and program accounts)
```

Comment:
```suggestion
                // Reset the account itself
```
It's not doing anything special for program accounts, so I think that's not relevant